src/main/java/fr/xephi/authme/command/executable/authme/RegisterAdminCommand.java

```diff
@@ -52,13 +51,13 @@ public void executeCommand(final CommandSender sender, List<String> arguments) {
         }
 
         bukkitService.runTaskOptionallyAsync(() -> {
-            if (dataSource.isAuthAvailable(playerNameLowerCase)) {
+            if (dataSource.isAuthAvailable(playerName)) {
                 commonService.send(sender, MessageKey.NAME_ALREADY_REGISTERED);
                 return;
             }
-            HashedPassword hashedPassword = passwordSecurity.computeHash(playerPass, playerNameLowerCase);
+            HashedPassword hashedPassword = passwordSecurity.computeHash(playerPass, playerName);
             PlayerAuth auth = PlayerAuth.builder()
-                .name(playerNameLowerCase)
+                .name(playerName)
                 .realName(playerName)
```

Comment:
```java
     .name(playerName)
     .realName(playerName)
```

This has bit of a funny feel now. It's absolutely 100% correct. I'm just wondering if it'd make sense to have a single `name(String)` function, since obviously from the realname we immediately know the all-lowercase name? This is maybe something we can look at after this pull request (for the sake of merging this soon). I expect that in some cases we do only want to set the `name` only (without a realname), but in most cases it would be nice if we just had to pass the name once and could tell the PlayerAuth builder "hey you, do what you need to do" :smile:

Reply:
yeah, we can enhance it in a future cleanup